### PR TITLE
fix(ctb): Add vm.assume to prevent clashes on additional test cases

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Addresses this CI [failure](https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/26959/workflows/1c632317-ecd9-4aa1-8e77-cb59eedd32a4/jobs/1256571). The issue it detecting was a known failure mode, but we were only filtering it out on one test case. 

This PR expands it, and also makes some incidental changes while I was in there.

### CODE REVIEW

File: packages/contracts-bedrock/test/DelayedVetoable.t.sol [hunk 1]
```diff
@@ -63,14 +74,15 @@ contract DelayedVetoable_Getters_TestFail is DelayedVetoable_Init {
         delayedVetoable.target();
         vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
         delayedVetoable.delay();
+        vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
+        delayedVetoable.queuedAt(keccak256(abi.encode(0)));
```
Comment (maurelian) [inline packages/contracts-bedrock/test/DelayedVetoable.t.sol:77]: these and the previous getter tests were missing

File: packages/contracts-bedrock/test/DelayedVetoable.t.sol [hunk 2]
```diff
@@ -63,14 +74,15 @@ contract DelayedVetoable_Getters_TestFail is DelayedVetoable_Init {
...
     }
 }
 
 contract DelayedVetoable_HandleCall_Test is DelayedVetoable_Init {
     /// @dev A call can be initiated by the initiator.
-    function testFuzz_handleCall_initiation_succeeds(bytes memory data) external {
-        assumeNonzeroData(data);
```
Comment (maurelian) [inline packages/contracts-bedrock/test/DelayedVetoable.t.sol:72]: yes, I did mean to remove `assumeNonzeroData()` here. Per the natspec I added on it above, it should only be used when the delay is not yet activated

File: packages/contracts-bedrock/test/DelayedVetoable.t.sol [hunk 3]
```diff
@@ -167,8 +183,13 @@ contract DelayedVetoable_HandleCall_TestFail is DelayedVetoable_Init {
     }
 
     /// @dev If the target reverts, it is bubbled up.
-    function testFuzz_handleCall_forwardingTargetReverts_reverts(bytes memory inData, bytes memory outData) external {
-        assumeNonzeroData(inData);
+    function testFuzz_handleCall_forwardingTargetReverts_reverts(
+        bytes calldata inData,
+        bytes calldata outData
```
Comment (maurelian) [inline packages/contracts-bedrock/test/DelayedVetoable.t.sol:187]: outdata could have still been `memory`, but that felt inconsistent.

File: packages/contracts-bedrock/test/DelayedVetoable.t.sol [hunk 4]
```diff
@@ -34,11 +34,21 @@ contract DelayedVetoable_Init is CommonTest {
...
+    /// @dev This function is used to prevent initiating the delay unintentionally.
+    ///      It should only be used on tests prior to the delay being activated.
     /// @param data The data to be used in the call.
     function assumeNonzeroData(bytes memory data) internal pure {
         vm.assume(data.length > 0);
     }
+
+    /// @dev This function is used to ensure that the data does not clash with the queuedAt function selector.
+    /// @param data The data to be used in the call.
+
```
Comment (refcell) [inline packages/contracts-bedrock/test/DelayedVetoable.t.sol]: nit
```suggestion
```

**Review by refcell — APPROVED** [review]

File: packages/contracts-bedrock/test/DelayedVetoable.t.sol (hunk 4 above)
Comment (refcell) [reply packages/contracts-bedrock/test/DelayedVetoable.t.sol]: Ah, this probably also fixes the ci lint failures. :pat: :pat: ci

**Pull request merged**